File: sdk/src/main/java/org/coralibre/android/sdk/internal/database/persistent/DaoDiagnosisKey.java
package org.coralibre.android.sdk.internal.database.persistent;

import androidx.room.Dao;
import androidx.room.Insert;
import androidx.room.OnConflictStrategy;
import androidx.room.Query;
import androidx.room.Update;

import org.coralibre.android.sdk.internal.database.persistent.entity.EntityDiagnosisKey;

import java.util.List;

@Dao
public interface DaoDiagnosisKey {

    @Insert(onConflict = OnConflictStrategy.ABORT)
    public void insertDiagnosisKeys(List<EntityDiagnosisKey> entityDiagnosisKeys);

    @Update(onConflict = OnConflictStrategy.REPLACE)
    public void updateDiagnosisKeys(List<EntityDiagnosisKey> entityDiagnosisKeys);

    @Query("SELECT * FROM EntityDiagnosisKey WHERE tokenString = :token")
    public List<EntityDiagnosisKey> getDiagnosisKeys(String token);


    // TODO Provide method to remove old keys (and also call it somewhere)


    @Query("DELETE FROM EntityDiagnosisKey")
    public void clearAllData();
}

File: sdk/src/main/java/org/coralibre/android/sdk/internal/database/persistent/DaoDiagnosisKey.kt
package org.coralibre.android.sdk.internal.database.persistent

import androidx.room.Dao
import androidx.room.Insert
import androidx.room.OnConflictStrategy
import androidx.room.Query
import androidx.room.Update
import org.coralibre.android.sdk.internal.database.persistent.entity.EntityDiagnosisKey

@Dao
interface DaoDiagnosisKey {
    // TODO what happens on update?
    @Insert(onConflict = OnConflictStrategy.ABORT)
    fun insertDiagnosisKeys(entityDiagnosisKeys: List<EntityDiagnosisKey>)

    @Update(onConflict = OnConflictStrategy.REPLACE)
    fun updateDiagnosisKeys(entityDiagnosisKeys: List<EntityDiagnosisKey>)

    @Query("SELECT * FROM EntityDiagnosisKey WHERE tokenString = :token")
    fun getDiagnosisKeys(token: String): List<EntityDiagnosisKey>
}
